docs/examples/Reinforcement_Learning_Sensor_Manager.py

```diff
@@ -0,0 +1,1002 @@
+
+        # Calculate a measurement from the sensor
+        measurement = set()
+        measurement |= self.sensor.measure(OrderedSet(truth[current_timestep] for truth in truths), noise=True)
+        hypotheses = data_associator.associate(self.tracks,
+                                               measurement,
+                                               current_timestep)
+
+        for track in self.tracks:
+            hypothesis = hypotheses[track]
+            if hypothesis.measurement:
+                post = updater.update(hypothesis)
+                track.append(post)
+            else:  # When data associator says no detections are good enough, we'll keep the prediction
+                track.append(hypothesis.prediction)
+
+        # Set the observation as the prior uncertainty of each target
+        observation = np.array(uncertainty, dtype=np.float32)
+
+        self.current_step += 1
+
+        if self.current_step >= self.max_episode_length-1:
+            self._episode_ended = True
+            return ts.termination(observation, reward)
+        else:
+            return ts.transition(observation, reward=reward, discount=1.0)
+
+    @staticmethod
+    def generate_action(action, tracks, sensor):
+        """This method is used to convert a tf-agents action into a Stone Soup action"""
+        for i, target in enumerate(tracks):
+            if i == action:
+                x_target = target.state.state_vector[0]-sensor.position[0]
+                y_target = target.state.state_vector[2]-sensor.position[1]
+                action_bearing = mod_bearing(np.arctan2(y_target, x_target))
+
+        action_generators = DwellActionsGenerator(sensor,
+                                                  attribute='dwell_centre',
+                                                  start_time=sensor.timestamp,
+                                                  end_time=sensor.timestamp+timedelta(seconds=1))
+
+        current_action = [action_generators.action_from_value(action_bearing)]
+        return current_action
+
+# Validate the environment to ensure that the environment returns the expected specs
+train_env = StoneSoupEnv()
+utils.validate_py_environment(train_env, episodes=5)
+
+# %%
+# RL Sensor Manager
+# -------------------------------------
+# To be able to use the RL environment we have designed, we need to make a ReinforcementSensorManager class, which
+# inherits from :class:`~.SensorManager`.
+# 
+# We introduce some additional methods that are used by tensorflow-agents: :func:`compute_avg_return`,
+# :func:`dense_layer`, and :func:`train`.
+# :func:`compute_avg_return` is used to find the average reward by using a given policy. This is used to evaluate the
+# training.
+# :func:`dense_layer` is used when generating the Q-Network, a neural network model that learns to predict Q-Values.
+# 
+# :func:`train` is used to generate the policy by running a large number of episodes through the Q-Network to work out
```

Comment:
Need to define 'episodes' here as it's a somewhat specialised term. Also, 1-sentence paragraph.